src/cc/api/BPF.cc

```diff
@@ -233,22 +233,39 @@ StatusTuple BPF::attach_usdt(const USDT& usdt, pid_t pid, int cpu,
       bool failed = false;
       std::string err_msg;
       int cnt = 0;
-      for (auto addr : u.addresses_) {
-        auto res =
-            attach_uprobe(u.binary_path_, std::string(), u.probe_func_, addr);
-        if (res.code() != 0) {
-          failed = true;
-          err_msg += "USDT " + u.print_name() + " at " + std::to_string(addr);
-          err_msg += ": " + res.msg() + "\n";
-          break;
+      for (auto bin_addr : u.bin_addresses_) {
+        bool break_out = false;
```

Comment:
Looks like bin_addresses_ only for debug purpose, right?

Reply:
Not entirely; it's used to keep track of the addresses to attach to. Basically instead of just an address, it now has a binary and an address.